### PR TITLE
Get jobs time window

### PR DESCRIPTION
**Pull request by natebrennand**
Adds filter options to limit jobs returned by `time_finished` value.

### CODE REVIEW

File: routes.coffee [hunk 1]
```diff
@@ -49,9 +49,11 @@ module.exports = (app) ->
 
   app.get '/api/jobs', auth.basic, (req, res) ->
     query =
```
Comment (azylman) [inline routes.coffee]: How is this different than `req.query`

Comment (natebrennand) [inline routes.coffee]: AFAICT it isn't except it makes the querystring parameters that are supported very explicit

File: test/db.coffee [hunk 2]
```diff
@@ -115,6 +115,64 @@ _.each [
           compare_jobs @jobs[1], jobs[0]
           done err
 
+      it "with an 'after' query returns only jobs from after the given time", (done) ->
+         db.delete_old_jobs new Date(), (err) ->
+          done err if err?
```
Comment (azylman) [inline test/db.coffee]: probably want to `return done err if err?`

File: test/db.coffee [hunk 3]
```diff
@@ -115,6 +115,64 @@ _.each [
...
+          # push in 'complete' jobs with time_finished values
+          @jobs = ((_.extend new Job "host:#{i}", "function#{i}", {time_finished: new Date i * 1000},
+              {state: 'complete'}) for i in [0..10])
+          add_jobs @jobs, (err) ->
+            done err if err?
```
Comment (azylman) [inline test/db.coffee]: `return`

File: test/db.coffee [hunk 4]
```diff
@@ -115,6 +115,64 @@ _.each [
...
+            db.get_jobs {after: 5000}, {page: 1, limit: 100}, (err, jobs, paging) =>
+              assert.ifError err
+              assert.equal jobs.length, 5  # 6,7,8,9,10
+              strictDeepEqual paging, {count: 5, current: 1, total: 1}
+              compare_jobs @jobs[6], jobs[0]
+              done null
+
+      it "with an 'after' query is higher than any existing 'time_finished' nothing is returned", (done) ->
+        db.delete_old_jobs new Date(), (err) ->
+          done err if err?
```
Comment (azylman) [inline test/db.coffee]: `return`

Comment (azylman) [inline test/db.coffee]: other places throughout this file

File: test/db.coffee [hunk 5]
```diff
@@ -115,6 +115,64 @@ _.each [
...
+
+      it "with a 'before' & 'after' will return a window of the finished jobs", (done) ->
+        db.delete_old_jobs new Date(), (err) ->
+          done err if err?
+          # push in 'complete' jobs with time_finished values
+          @jobs = ((_.extend new Job "host:#{i}", "function#{i}", {time_finished: new Date i * 1000},
+              {state: 'complete'}) for i in [0..10])
+          add_jobs @jobs, (err) ->
+            return done err if err?
+            db.get_jobs {after: 0, before: 6000}, {page: 1, limit: 100}, (err, jobs, paging) =>
```
Comment (azylman) [inline test/db.coffee]: With an `after` of 0, this is equivalent to `before: 6000`, right? Maybe actually use an after value so it's limited on both sides

File: routes.coffee (hunk 1 above)
Comment (azylman) [inline routes.coffee]: Could use pick for that if explicit is the goal: `query = _.pick req.query, 'state', 'function', 'where', 'after', 'before'`

**Pull request merged**